app/controllers/api2_controller.rb

```diff
@@ -143,15 +143,20 @@ def upload_from_request_body
   end
 
   def render_api_results(args)
+    @user = user_from_key(args[:api_key])
     @api = API2.execute(args)
-    User.current = @user = @api.user
+    User.current = @api.user
```

Comment:
I'm gathering that in the API, sounds like you still need to define `User.current` at this point.

Reply:
Updated and it looks good.